lib/backend/k8s/k8s.go

```diff
@@ -210,6 +211,33 @@ func NewKubeClient(ca *apiconfig.CalicoAPIConfigSpec) (api.Client, error) {
 	return kubeClient, nil
 }
 
+// deduplicate removes any duplicated values and returns a new slice, keeping the order unchanged
+func deduplicate(s []string) []string {
+	encountered := map[string]bool{}
+	ret := make([]string, 0)
+	for i := range s {
+		if encountered[s[i]] {
+			continue
+		}
+		encountered[s[i]] = true
+		ret = append(ret, s[i])
+	}
+	return ret
+}
+
+// fill out loading rules based on filename(s) encountered in specified kubeconfig
+func fillLoadingRulesFromKubeConfigSpec(loadingRules *clientcmd.ClientConfigLoadingRules, kubeConfig string) {
```

Comment:
Looks sensible.  I guess the main reason to copy this from client-go's NewDefaultClientConfigLoadingRules() is to maintain our slightly different environment variable setup?

Reply:
Indeed. Almost all of our env vars are setup under calicoctl and passed down as `*apiconfig.CalicoAPIConfigSpec` that would have all the env vars and command line arguments already pre-combined